src/test/java/com/epam/test_generator/controllers/CaseControllerTest.java

```diff
@@ -431,9 +468,63 @@ public void testRemoveCase_return500whenRuntimeException() throws Exception {
         doThrow(RuntimeException.class).when(casesService).removeCase(anyLong(), anyLong());
 
         mockMvc.perform(delete("/suits/" + SIMPLE_SUIT_ID + "/cases/" + SIMPLE_CASE_ID))
-                .andDo(print())
-                .andExpect(status().isInternalServerError());
+            .andDo(print())
+            .andExpect(status().isInternalServerError());
 
         verify(casesService).removeCase(eq(SIMPLE_SUIT_ID), eq(SIMPLE_CASE_ID));
     }
+
+    @Test
+    public void testRemoveCases_return200whenRemoveCases() throws Exception {
+
+        mockMvc.perform(delete("/suits/" + SIMPLE_SUIT_ID + "/cases")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(mapper.writeValueAsString(CASE_IDS)))
+            .andDo(print())
+            .andExpect(status().isOk());
+
+        verify(casesService).removeCases(eq(SIMPLE_SUIT_ID), eq(Arrays.asList(CASE_IDS)));
+    }
+
+    @Test
+    public void testRemoveCases_return404whenSuitNotExistOrCaseNotExist() throws Exception {
+        doThrow(NotFoundException.class).when(casesService).removeCases(anyLong(),
+            anyListOf(Long.class));
+
+        mockMvc.perform(delete("/suits/" + SIMPLE_SUIT_ID + "/cases")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(mapper.writeValueAsString(CASE_IDS)))
+            .andDo(print())
+            .andExpect(status().isNotFound());
+
+        verify(casesService).removeCases(eq(SIMPLE_SUIT_ID), eq(Arrays.asList(CASE_IDS)));
+    }
+
+    @Test
+    public void testRemoveCases_return400whenSuitNotContainsCase() throws Exception {
+        doThrow(BadRequestException.class).when(casesService)
+            .removeCases(anyLong(), anyListOf(Long.class));
+
+        mockMvc.perform(delete("/suits/" + SIMPLE_SUIT_ID + "/cases")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(mapper.writeValueAsString(CASE_IDS)))
+            .andDo(print())
+            .andExpect(status().isBadRequest());
+
+        verify(casesService).removeCases(eq(SIMPLE_SUIT_ID), eq(Arrays.asList(CASE_IDS)));
+    }
+
+    @Test
+    public void testRemoveCases_return500whenRuntimeException() throws Exception {
+        doThrow(RuntimeException.class).when(casesService)
+            .removeCases(anyLong(), anyListOf(Long.class));
+
+        mockMvc.perform(delete("/suits/" + SIMPLE_SUIT_ID + "/cases")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(mapper.writeValueAsString(CASE_IDS)))
+            .andDo(print())
+            .andExpect(status().isInternalServerError());
+
+        verify(casesService).removeCases(eq(SIMPLE_SUIT_ID), eq(Arrays.asList(CASE_IDS)));
+    }
```

Comment:
Add success and failure cases for **containsAll** check.

Reply:
Added tests that drops on containsAll check, and with duplicate ids